chessengine/bitboard.py

```diff
@@ -724,34 +727,8 @@ def play_pvp(self) -> None:
             print(self)
             lines_printed = 11
 
-            input_accepted = False
-            while not input_accepted:
-                print(last_move)
-                move = input(
-                    f"Enter the move you want to make ({side_to_move.capitalize()}'s turn) - "
-                ).strip()
-                lines_printed += 2
-
-                if move.lower() == "q":
-                    print("Thanks for playing!")
-                    return
-                if move.lower() == "u":
-                    try:
-                        self.undo_move()
-                    except RuntimeError:
-                        print("No moves have been made yet to undo!")
-                        lines_printed += 1
-                    break
+            move, lines_added, _ = self.handle_player_move(side_to_move, last_move)
```

Comment:
Here we should add a check to see if a move was undone and continue if so, just like in `play` -
```python
if move_undone:
    continue
```

Reply:
Ah yes. That would be required with the pvp undo change.